docs/vue.md

```diff
@@ -20,118 +20,318 @@ To get started, load either the production or development version of Vue in your
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
+
+```markdown
+<div id="counter">
+  <button @click="count -= 1">-</button>
+  {{ count }}
+  <button @click="count += 1">+</button>
+</div>
+```
+
+<output id="counter">
+  <button @click="count -= 1">-</button>
+  {{ count }}
+  <button @click="count += 1">+</button>
+</output>
+
+## Components
+
+Use `vueComponents` to register Vue components using the component name as the key with an object containing Vue options as the value. Component `data` is unique for each instance and will not persist as users navigate the site.
+
+```js
+window.$docsify = {
+  vueComponents: {
+    'button-counter': {
+      template: `
+        <button @click="count += 1">
+          You clicked me {{ count }} times
+        </button>
+      `,
+      data() {
+        return {
+          count: 0,
+        };
+      },
+    },
+  },
+};
+```
 
-<div id="example-1">
-  <p>{{ message }}</p>
+```markdown
+<button-counter></button-counter>
+<button-counter></button-counter>
+```
 
-  <button v-on:click="hello">Say Hello</button>
+<output data-lang="output">
+  <button-counter></button-counter>
+  <button-counter></button-counter>
+</output>
 
-  <button v-on:click="counter -= 1">-</button>
-  {{ counter }}
-  <button v-on:click="counter += 1">+</button>
-</div>
+## Markdown script
 
-<!-- prettier-ignore-end -->
+Vue content can mounted using a `<script>` tag in your markdown pages.
 
-!> Only the first `<script>` tag in a markdown file is executed. If you are working with multiple Vue components, all `Vue` instances must be created within this tag.
+!> Only the first `<script>` tag in a markdown file is executed. If you wish to mount multiple Vue instances using a script tag, all instances must be mounted within the first script tag in your markdown.
 
+```html
+<!-- Vue 2.x  -->
 <script>
   new Vue({
-    el: "#example-1",
-    data: function() {
-      return {
-        counter: 0,
-        message: "Hello, World!"
-      };
-    },
-    methods: {
-      hello: function() {
-        alert(this.message);
-      }
-    }
+    el: '#example',
+    // Options...
   });
 </script>
+```
+
+```html
+<!-- Vue 3.x  -->
+<script>
+  Vue.createApp({
+    // Options...
+  }).mount('#example');
+</script>
+```
+
+## Technical Notes
+
+- Docsify processes Vue content in the following order:
+  1. Execute markdown `<script>`
+  1. Register `vueComponents`
+  1. Mount DOM elements via `vueMounts`
+  1. Process unmounted Vue content using `vueGlobalOptions`
+- Docsify will not mount an existing Vue instance or an element that contains an existing Vue instance.
+- Docsify will automatically destroy/unmount all Vue instances it creates before new page content is loaded.
+- When processing `vueGlobalOptions`, docsify parses the child elements within the main content area (`#main`) and mounts the element if it contains Vue content. Docsify does not parse each individual node within the main content area.
```

Comment:
I'm not clear on the last bullet point. I guess if it'll start to explain this magic part, it might just need to go into more detail.

Reply:
Yes, I agree this description could use some work. To be honest, I debated about even adding it. I've updated the description to (hopefully) make it more clear. At the end of the day, this text exists to help explain why this edge case won't work:

```js
window.$docsify = {
  vueGlobalOptions: {
    data() {
      return {
        globalMsg: 'Hello from vueGlobalOptions!',
      };
    },
  },
  vueMounts: {
    '#test': {
      data() {
        return {
          mountMsg: 'Hello from vueMounts!'
        };
      },
    },
  },
};
```

**Example 1:** This will work

```html
<!-- Source -->
<p>{{ globalMsg }}</p>
<p id="test">{{ mountMsg }}</p>
```
```html
<!-- Output -->
<p>Hello from vueGlobalOptions!</p>
<p id="test">Hello from vueMounts!</p>
```

In this example, docsify first mounts the top-level  `<p id="test">` from `vueMounts`. Docsify then iterates over the top-level child elements (`<p>`), skips the ones that contain or are themselves Vue instances, detects Vue template syntax in the top-level `<p>` element, and mounts it using `vueGlobalOptions`. Great.

**Example 2:** This will _not_ work

```html
<!-- Source -->
<div>
  <p>{{ globalMsg }}</p>
  <p id="test">{{ mountMsg }}</p>
</div>
```
```html
<!-- Output -->
<div>
  <p>{{ globalMsg }}</p>
  <p id="test">Hello from vueMounts!</p>
</div>
```

Like the previous example, docsify first mounts the top-level  `<p id="test">` from `vueMounts`. Docsify then iterates over the top-level child elements (`<div>`), detects that `<div>` contains a Vue instance (`<p id="test">`) and skips that element. This behavior is explained in the following "Technical Notes" bullet point:

> - Docsify will not mount an existing Vue instance or an element that contains an existing Vue instance.

I think it is highly unlikely that this will be an issue, but I added the related technical notes in hopes of making life easier for everyone if it does. The trick is how to do this succinctly without going into the same level of detail above. Hopefully the changes I've put in place are good enough for now. We can always revisit if/when needed.

Reply:
@jhildenbiddle The example of what works vs what doesn't will be very helpful to some people. I think it'd be worth pasting what you wrote into the docs.